File: chef/hmappy2.cpp
#include<bits/stdc++.h>
using namespace std;
long long int gcd(long long int a,long long int b)
{
    if(a==0)
        return b;
    if(b==0)
        return a;
    if(a>b)
        return gcd(a%b,b);
    else
        return gcd(a,b%a);
}
int main(int argc, char const *argv[]) {
    ios_base::sync_with_stdio(false);
    int t;
    cin>>t;
    while(t--)
    {
        long long int n,k,a,b,res,g;
        cin>>n>>a>>b>>k;
        g=gcd(a,b);
        g=(a*b)/g;
        res=n/a+n/b-2*(n/g);
        if(res<k)
            cout<<"Lose\n";
        else
            cout<<"Win\n";
    }
    return 0;
}

File: chef/hmappy2.py
def gcd(a,b):
    if(a==0):
        return b
    if(b==0):
        return a
    if(a>b):
        return gcd(a%b,b)
    else:
        return gcd(a,b%a)

t=int(input())
for i in range(t):
    ar = list(map(int,raw_input().split()))
    x  = ar[0]*gcd(ar[1],ar[2])/(ar[1]*ar[2])
    res=ar[0]/ar[1]+ar[0]/ar[2]-2*x
    print(res)
    if(res<ar[3]):
        print("Lose")
    else:
        print("Win")
